tests/baselines/reference/moduleResolutionWithExtensions_unexpected2.trace.json

```diff
@@ -2,30 +2,25 @@
     "======== Resolving module 'foo' from '/a.ts'. ========",
     "Module resolution kind is not specified, using 'NodeJs'.",
     "Loading module 'foo' from 'node_modules' folder, target file type 'TypeScript'.",
-    "File '/node_modules/foo.ts' does not exist.",
-    "File '/node_modules/foo.tsx' does not exist.",
-    "File '/node_modules/foo.d.ts' does not exist.",
     "Found 'package.json' at '/node_modules/foo/package.json'.",
     "'package.json' does not have a 'typings' field.",
     "'package.json' has 'types' field 'foo.js' that references '/node_modules/foo/foo.js'.",
     "File '/node_modules/foo/foo.js' exist - use it as a name resolution result.",
     "File '/node_modules/foo/foo.js' has an unsupported extension, so skipping it.",
-    "Loading module as file / folder, candidate module location '/node_modules/foo/foo.js', target file type 'TypeScript'.",
+    "Loading module as file / folder, candidate module location '/node_modules/foo/foo.js', target file type 'TypeScript ('.d.ts' preferred)'.",
+    "File '/node_modules/foo/foo.js.d.ts' does not exist.",
     "File '/node_modules/foo/foo.js.ts' does not exist.",
     "File '/node_modules/foo/foo.js.tsx' does not exist.",
-    "File '/node_modules/foo/foo.js.d.ts' does not exist.",
     "File name '/node_modules/foo/foo.js' has a '.js' extension - stripping it.",
+    "File '/node_modules/foo/foo.d.ts' does not exist.",
     "File '/node_modules/foo/foo.ts' does not exist.",
     "File '/node_modules/foo/foo.tsx' does not exist.",
-    "File '/node_modules/foo/foo.d.ts' does not exist.",
     "Directory '/node_modules/foo/foo.js' does not exist, skipping all lookups in it.",
+    "File '/node_modules/foo/index.d.ts' does not exist.",
     "File '/node_modules/foo/index.ts' does not exist.",
     "File '/node_modules/foo/index.tsx' does not exist.",
-    "File '/node_modules/foo/index.d.ts' does not exist.",
     "Directory '/node_modules/@types' does not exist, skipping all lookups in it.",
     "Loading module 'foo' from 'node_modules' folder, target file type 'JavaScript'.",
-    "File '/node_modules/foo.js' does not exist.",
-    "File '/node_modules/foo.jsx' does not exist.",
```

Comment:
removal of this path lookup seems incorrect.
Refer https://nodejs.org/api/modules.html

> 
> LOAD_NODE_MODULES(X, START)
> 1. let DIRS=NODE_MODULES_PATHS(START)
> 2. for each DIR in DIRS:
>    **a. LOAD_AS_FILE(DIR/X)**
>    b. LOAD_AS_DIRECTORY(DIR/X)

It looks like node does look for node_modules/foo.js